### ISSUE TITLE
Infinite loop in subtract

### ISSUE THREAD

**Issue by bernizt** [alexbol99/flatten-boolean-op#11]
I have encountered a problem with the **subtract** function

```
const area = new Polygon();
area.addFace([new Point(612.31281198, 49.2512479201),
new Point(490.31281198, 207.25124792),
new Point(257.31281198, 207.25124792),
new Point(161.31281198, 98.2512479201),
new Point(49.31281198, 207.25124792),
new Point(49.31281198, 629.25124792),
new Point(237.31281198, 536.25124792),
new Point(490.31281198, 536.25124792),
new Point(817.31281198, 629.25124792)]);

const borderWide = new Polygon();
borderWide.addFace([new Point(49.31281198, 629.25124792),
new Point(237.31281197999985, 536.25124792),
new Point(172.8105605334842, 536.25124792),
new Point(49.31281198, 597.3432192789045)]);

// The subtract operation seems to loop indefinitely in 1.1.4
subtract(area, borderWide);
```

You can also test it here:
[runkit.com/bernizt/bug-in-flatten-js-boolean-op-subtract](https://runkit.com/bernizt/bug-in-flatten-js-boolean-op-subtract)

**Comment by alexbol99** [comment]
Hello @bernizt 

Bug was in intersection module of  https://github.com/alexbol99/flatten-js package.
Bug fixed in version 1.0.18
See live demo in the notebook: https://observablehq.com/d/b7e51bc17d8281a5

Thank you for cooperation,
Alex

**Comment by bernizt** [comment]
Hello @alexbol99 
Thank you very much for your fast reaction, the bug seems to be indeed gone in flatten-js version 1.0.18.
